File: agent/OuroBuild.Agent/Models/AgentRegistrationRequest.cs
using System.Text.Json.Serialization;

namespace OuroBuild.Agent.Models;

public sealed record AgentRegistrationRequest(
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("machine_name")] string MachineName,
    [property: JsonPropertyName("user_name")] string UserName,
    [property: JsonPropertyName("version")] string Version);

File: app/models/agent/agent_registration_request.py
"""
--------------------------------------------------------------------
Projeto : OuroBuild
Arquivo : agent_registration_request.py
Descrição : Dados enviados pelo Agent durante o registro.
--------------------------------------------------------------------
"""

from pydantic import BaseModel, Field


class AgentRegistrationRequest(BaseModel):
    """Representa os dados de registro de um Agent."""

    name: str = Field(min_length=1, max_length=100)
    machine_name: str = Field(min_length=1, max_length=255)
    user_name: str = Field(min_length=1, max_length=255)
    version: str = Field(min_length=1, max_length=50)
